### PR TITLE
Threat modelling framework

### PR DESCRIPTION
**Pull request by UlfNorell**
Updates the quickcheck-contractmodel dependency to the latest version with the new threat modelling framework.

- Reenable double satisfaction tests (this now comes with the threat modelling)
- Some simple threat model examples (GameStateMachine and Escrow)
- Use min Ada balance change logic from quickcheck-contractmodel

### CODE REVIEW

File: plutus-use-cases/plutus-use-cases.cabal [hunk 1]
```diff
@@ -224,6 +224,38 @@ test-suite plutus-use-cases-test
...
+    , cardano-ledger-shelley
+    , cardano-ledger-shelley-ma
+    , cardano-slotting
+    , ouroboros-consensus
+    , ouroboros-consensus-cardano
+    , plutus-ledger-api
+    , plutus-tx
+    , strict-containers
+    , time
+
```
Comment (koslambrou) [inline plutus-use-cases/plutus-use-cases.cabal:258]: These should go inside one of the categories above.

File: plutus-use-cases/test/Spec/Auction.hs [hunk 2]
```diff
@@ -358,9 +367,9 @@ check_propAuctionWithCoverage = do
         (set minLogLevel Critical options) covopts (const (pure True))
   writeCoverageReport "Auction" cr
 
--- TODO: re-activate when thread modelling framework is merged
--- prop_doubleSatisfaction :: Actions AuctionModel -> Property
--- prop_doubleSatisfaction = checkDoubleSatisfactionWithOptions options defaultCoverageOptions
+-- | Note: this property fails, because the contract is vulnerable to double satisfaction.
```
Comment (koslambrou) [inline plutus-use-cases/test/Spec/Auction.hs:370]: If it fails, it shouldn't pass CI right?

**Review by koslambrou — APPROVED** [review]
LGTM. Conflicts need to be fixed, I'll merge once that's done :) Sorry for the delay.

File: plutus-use-cases/test/Spec/Auction.hs (hunk 2 above)
Comment (UlfNorell) [reply plutus-use-cases/test/Spec/Auction.hs:370]: It included in the test suite under an `expectFailure` (see below)`, so CI passes only when it *does* fail.

Comment (koslambrou) [reply plutus-use-cases/test/Spec/Auction.hs:370]: Oh missed it. Thanks

**Pull request merged**